### ISSUE TITLE
Strange startup behavior in Lighty RNC App

### ISSUE THREAD

**Issue by ercankilic** [PANTHEONtech/lighty#881]
Hi colleagues,

We are using lighty 15.1.0 version in our project. From time to time, we face an issue which lighty.io refuses connection although it seems the docker container running it is up and running. While we are investigating the issue, we found out that, sometimes although initialization fails, lighty continues start up procedure.

Part of what we see from initialization log is below:

```
ERROR [pool-1-thread-1] (RncLightyModule.java:86) - Unable to initialize and start RNC lighty.io module!
io.lighty.applications.rnc.module.exception.RncLightyAppStartException: Exception was thrown during initialization of lighty.io module (class io.lighty.core.controller.impl.LightyControllerImpl)!
	at io.lighty.applications.rnc.module.RncLightyModule.startAndWaitLightyModule(RncLightyModule.java:161)
	at io.lighty.applications.rnc.module.RncLightyModule.initProcedure(RncLightyModule.java:69)
	at io.lighty.core.controller.api.AbstractLightyModule.lambda$start$0(AbstractLightyModule.java:119)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.TimeoutException: Waited 60 seconds (plus 8116241 nanoseconds delay) for TrustedListenableFutureTask@794c96c0[status=PENDING, info=[task=[running=[RUNNING ON pool-4-thread-1], io.lighty.core.controller.api.AbstractLightyModule$$Lambda$92/0x00000008401dc440@5f2447a7]]]
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:508)
	at com.google.common.util.concurrent.FluentFuture$TrustedFuture.get(FluentFuture.java:95)
	at io.lighty.applications.rnc.module.RncLightyModule.startAndWaitLightyModule(RncLightyModule.java:152)
	... 8 more
 INFO [pool-1-thread-1] (AbstractLightyModule.java:121) - LightyModule RncLightyModule started.
```

What we trace from code based on this log is, in package io.lighty.applications.rnc.app.Main:102 code tries to call rncLightyModule.start.get() and this method returns boolean which is false when start-up fails. But since this return value is not handled (and also the exception thrown from RncLightyModule which is RncLightyAppStartException is not handled neither), startup procedure continues although initialization failed.

Please kindly feedback about the issue.

Kind regards,

Ercan

**Comment by PeterSuna** [comment]
Hi ercankilic,
My apologies for the late reply. Your reported issue should be now resolved. Currently, should be fixed closing lighty application when some problem in application startup is occurred. From provided logs, I can assume that initializing lighty core with your configuration takes more time than application expect. I have added an option to increase time-out for modules inside
RNC application. https://github.com/PANTHEONtech/lighty/blob/15.1.x/lighty-applications/lighty-rnc-app-aggregator/README.md?plain=1#L43
Now you can increase module time-out and your problem could be resolved or the real issue why it takes longer will be shown.
Soon this changes will be included in lighty 15.2.0 release.
 
 Thanks for feedback

**Comment by ihrasko** [comment]
Fixed in Lighty.io 15.2.0.

Enjoy!
